src/xrCore/Animation/Bone.hpp

```diff
@@ -465,60 +455,61 @@ class XRCORE_API CBoneData : public IBoneData
+    [[nodiscard]] IC u16 GetParentID() const override { return ParentID; }
     // assign face
     void AppendFace(u16 child_idx, u16 idx) { child_faces[child_idx].push_back(idx); }
     // Calculation
     void CalculateM2B(const Fmatrix& Parent);
 
 private:
-    IBoneData& GetChild(u16 id) { return *children[id]; }
-    const IBoneData& GetChild(u16 id) const { return *children[id]; }
-    u16 GetNumChildren() const { return (u16)children.size(); }
-    const SJointIKData& get_IK_data() const { return IK_data; }
-    const Fmatrix& get_bind_transform() const { return bind_transform; }
-    const SBoneShape& get_shape() const { return shape; }
-    const Fobb& get_obb() const { return obb; }
-    const Fvector& get_center_of_mass() const { return center_of_mass; }
-    float get_mass() const { return mass; }
-    virtual u16 get_game_mtl_idx() const override { return game_mtl_idx; }
-    virtual shared_str GetMaterialName() const override { return name; }
-    float lo_limit(u8 k) const { return IK_data.limits[k].limit.x; }
-    float hi_limit(u8 k) const { return IK_data.limits[k].limit.y; }
+    IBoneData& GetChild(u16 id) override { return *children[id]; }
+    [[nodiscard]] const IBoneData& GetChild(u16 id) const override { return *children[id]; }
+    [[nodiscard]] u16 GetNumChildren() const override { return children.size(); }
+    [[nodiscard]] const SJointIKData& get_IK_data() const override { return IK_data; }
+    [[nodiscard]] const Fmatrix& get_bind_transform() const override { return bind_transform; }
+    [[nodiscard]] const SBoneShape& get_shape() const override { return shape; }
+    [[nodiscard]] const Fobb& get_obb() const override { return obb; }
+    [[nodiscard]] const Fvector& get_center_of_mass() const override { return center_of_mass; }
+    [[nodiscard]] float get_mass() const override { return mass; }
+    [[nodiscard]] u16 get_game_mtl_idx() const override { return game_mtl_idx; }
+    [[nodiscard]] shared_str GetMaterialName() const override { return name; }
+    [[nodiscard]] float lo_limit(u8 k) const override { return IK_data.limits[k].limit.x; }
+    [[nodiscard]] float hi_limit(u8 k) const override { return IK_data.limits[k].limit.y; }
 
 public:
-    virtual size_t mem_usage()
+    virtual std::size_t mem_usage()
     {
-        size_t sz = sizeof(*this) + sizeof(vecBones::value_type) * children.size();
-        for (auto c_it = child_faces.begin(); c_it != child_faces.end(); ++c_it)
-            sz += c_it->size() * sizeof(FacesVec::value_type) + sizeof(*c_it);
+        std::size_t sz = sizeof(*this) + sizeof(vecBones::value_type) * children.size();
+        for (const auto& c : child_faces)
+        {
+            sz += c.size() * sizeof(FacesVec::value_type) + sizeof(c);
+        }
         return sz;
     }
 };
 
-enum EBoneCallbackType
+enum EBoneCallbackType : u32
 {
-    bctDummy = u32(0), // 0 - required!!!
+    bctDummy = 0, // required!!!
```

Comment:
We don't even need that now 😉 
```suggestion
    bctDummy,
```

Reply:
Right, removed.